vispy/color/colormap.py

```diff
@@ -440,6 +441,24 @@ def texture_lut(self):
         return texture_LUT
 
 
+class MatplotlibColormap(Colormap):
+    """Use matplotlib colormaps if installed.
+
+    Parameters
+    ----------
+    name : string
+        Name of the colormap.
+    """
+
+    def __init__(self, name):
+        from matplotlib.cm import ScalarMappable, cmap_d
+        assert has_matplotlib()
```

Comment:
This assertion is happening after the import. You could move it up a line or would it be cleaner to change the if statement below to be `if name in _colormaps; elif has_matplotlib(); else: KeyError(...)`

Reply:
also `assert` statements are skipped when running Python in optimized code so it shouldn't be used as an actual check that you want to be guaranteed to run

Reply:
Oh yes, sorry for the mistake. @larsoner what's Python optimized code? So finally I should remove `assert` in the `MatplotlibColormap`?

Reply:
running with `python -O`

I would change the code to:
```
if not has_matplotlib():
    raise ImportError('informative message')
from matplotlib.cm ...
```

Reply:
In `get_colormap` @djhoese suggest something like `if name in _colormaps: ... elif has_matplotlib():...` so the `if not has_matplotlib(): raise ImportError('informative message')` inside `MatplotlibColormap` isn't really useful?

Reply:
The only clean way I see right now to avoid matplotlib error message is that  `get_colormaps` return colormap name of both vispy and mpl if mpl is installed

It should looks like :
```py
if name not in get_colormaps():
    raise KeyError('colormap name %s not found' % name)
if name in _colormaps:  # vispy cmap
    cmap = _colormaps[name]
elif has_matplotlib():  # matplotlib colormap
    cmap = MatplotlibColormap(name)
```

Reply:
Are you calling `get_colormaps` inside `get_colormaps`? I think I prefer my original suggestion with the `if name in _colormaps; elif has_matplotlib(); else: raise KeyError`.

Reply:
That seems fine to me. And you probably don't really even need the check I suggested because the ImportError that will be raised just by the nested mpl import is probably sufficient

Reply:
No, I'm not calling `get_colormaps`

Reply:
ok for the `if name in _colormaps; elif has_matplotlib(); else: raise KeyError`. In case of a typo, the default Matplotlib error message is going to be raised, but not the one of vispy

Reply:
I'm ok with that because I think any other way would be putting mpl logic outside of the colormap class. I didn't really understand your code above where you put `if name not in get_colormaps()`.